api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -107,6 +116,28 @@ type XForwardedForSettings struct {
 	NumTrustedHops *uint32 `json:"numTrustedHops,omitempty"`
 }
 
+// CustomHeader provides configuration for determining the client IP address for a request based on
+// a trusted custom HTTP header. This uses the the custom_header original IP detection extension.
+// Refer to https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/original_ip_detection/custom_header/v3/custom_header.proto
+// for more details.
+type CustomHeaderExtensionSettings struct {
+	// HeaderName of the of the header containing the original downstream remote address, if present.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=255
+	// +kubebuilder:validation:Pattern="^[A-Za-z0-9-]+$"
+	//
+	HeaderName string `json:"headerName"`
+	// RejectWithStatus is the HTTP response status to use when detection fails, if present. May be
+	// any valid HTTP response status code within the range 400-511 (inclusive).
+	//
+	// +kubebuilder:validation:Minimum=400
+	// +kubebuilder:validation:Maximum=511
+	//
+	// +optional
+	RejectWithStatus *uint32 `json:"rejectWithStatus,omitempty"`
```

Comment:
instead of this, should this be ?
```
failClosed:
  status
```

Reply:
I like `failClosed` and keeping the terms uniform here. Would it make sense to make it flatter so `failClosed` is similarly boolean?

```yaml
failClosed: bool       # default false
failStatus: uint32     # default 400
```

Reply:
Might be ok to just go with a single boolean rejecting with 400 bad request when `failClosed: true` instead of making the status code configurable here as well. Wdyt?

Reply:
`failClosed: bool` with a default status sgtm, from the docs, looks like the preferred default is `403` 
https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/original_ip_detection/custom_header/v3/custom_header.proto

Reply:
403 sounds good, let's go with it